language-server/vscoqtop/lspManager.ml

```diff
@@ -277,6 +277,21 @@ let coqtopStepForward ~id params : (string * Dm.DocumentManager.events) =
       |> List.flatten)] in
       output_json @@ mk_response ~id ~result
 
+let coqtopGetDeclarationLocation ~id params =
+  let open Yojson.Basic.Util in
+  let textDocument = params |> member "textDocument" in
+  let uri = textDocument |> member "uri" |> to_string in
+  let loc = params |> member "position" |> parse_loc in
+  let requestedDeclaration = params |> member "requestedDeclaration" |> to_string in
+  let st = Hashtbl.find states uri in
+  match Dm.DocumentManager.get_location st loc requestedDeclaration with
+  | None -> ()
+  | Some (path, range) ->
+    let result = `Assoc [
+      "path", `String path;
+    ] in
```

Comment:
We will need to remember to add `range` here at some point (or just add it to the assoc-list based on whether it is Some or None).

Reply:
yes